cmd/sonobuoy/app/gen_test.go

```diff
@@ -205,6 +205,17 @@ func TestGetConfig(t *testing.T) {
 				},
 				PluginSearchPath: defaultPluginSearchPath,
 			},
+		}, {
+			name:     "Manually specified plugins should result in empty selection",
+			input:    &genFlags{},
+			cliInput: "--plugin e2e",
+			expected: &config.Config{
+				Namespace:        "heptio-sonobuoy",
+				WorkerImage:      "gcr.io/heptio-images/sonobuoy:" + buildinfo.Version,
+				ImagePullPolicy:  "IfNotPresent",
+				PluginSelections: nil,
```

Comment:
Empty selection now means "run every plugin in the search path". Did not solve the "run no plugins, only queries" issue.

I could:
 1) not worry about it as that is a _very_ small group if anyone does it and they can always run `--mode quick` without much of a meaningful slowdown.
 2) Maybe the nil slice indicates run all, the empty slice means run nothing? I hate to split hairs there but that is kind of the point of the nil value.

Reply:
So this [PR][https://github.com/heptio/sonobuoy/pull/684] made nil -> run all and empty slice means run nothing.

To order to actually execute that path you'd have to edit the YAML manually. I'm fine with this since it we'd be making the common cases easier without completely preventing the rare use case.